middleware/RetryHandler.md

```diff
@@ -10,7 +10,11 @@ Provide a reusuable component that provides application developers with effectiv
 - A `retry-attempt` header SHOULD be added which contains a numeric value representing the retry number.  This value will be used for diagnostics and determining the effectiveness of the retry handler.
 - Retries MUST respect the `retry-after` response header if provided.
 - Where no `retry-after` header is provided by the server, an exponential backoff with random offset hueristic should be used to determine the retry delay.
-- Retries should be limited to a maximum value.
+- Retries should be limited to a maximum delay value. The default value for this is set at 180 seconds.
+- The client code can specify a custom value for the maximum delay.
+- Retries will be based on the cumulative retry delays compared against the maximum delay specified by the client code, cumulatively added across all retries within the context of a single call. This is applicable to both 429 and 503/504.
+- Upon expiry of the maximum delay, when the cumulative retry delay is greater than the specified maximum delay, the `Task` should be cancelled and an `exception` thrown with a relevant message.
```

Comment:
I don't think we should be throwing exception because we exceeded the RetriesTimeLimit.  We should simply not retry again.  We could try and cancel a currently running task, but I'm not sure this is wise.  This would could cause the server to do a bunch of work that is then wasted.

The goal of this property should be to set a limit, that if exceeded prevents additional retries.  We should also account for exceeding the limit while waiting to retry, which is covered in the next point.